src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```
@@ -345,54 +430,84 @@ private void resubscribe(boolean sessionPresent) {
+                                                                } else {
+                                                                    logger.atError()
+                                                                            .event(RESUB_LOG_EVENT)
+                                                                            .cause(error)
+                                                                            .kv(
+                                                                                    TOPIC_KEY,
+                                                                                    entry.getKey())
+                                                                            .log(
+                                                                                    "Failed to subscribe to topic. Will retry later");
+                                                                }
+                                                            }));
+                                }
+                                // Block and wait for all subscriptions to finish
+                                CompletableFuture<?> allSubFutures =
+                                        CompletableFuture.allOf(
+                                                subFutures.toArray(new CompletableFuture[0]));
+                                try {
+                                    allSubFutures.get();
+                                } catch (InterruptedException e) {
+                                    logger.atWarn()
+                                            .event(RESUB_LOG_EVENT)
+                                            .cause(e)
+                                            .log(
+                                                    "Subscription interrupted. Cancelling subscriptions");
+                                    allSubFutures.cancel(true);
+                                } catch (ExecutionException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Swallowing exceptions without logging can make it difficult to identify and troubleshoot issues, potentially leading to application instability, hidden bugs, or unexpected behavior. To avoid this, ensure that all exceptions are logged using appropriate logging levels. When logging exceptions, include the exception object or its stack trace to capture detailed information. For critical exceptions, rethrow them when necessary to provide sufficient context for effective troubleshooting.

Suggested remediation:
Handle all exceptions by creating a log for each exception using the message thrown.
```
@@ -493,2 +493,3 @@
                                     // Do nothing. Errors already handled in individual subscription
+                                    log.error(&#34;ExecutionException is caught.&#34;, e);
                                     // future&#39;s whenComplete stage

```